zcash_primitives/src/sapling.rs

```diff
@@ -471,12 +471,7 @@ impl Note {
     pub(crate) fn generate_or_derive_esk_internal<R: RngCore>(&self, rng: &mut R) -> jubjub::Fr {
         match self.derive_esk() {
             None => {
```

Comment:
```suggestion
            None => jubjub::Fr::random(rng),
```

(GitHub won't let me do this as a single suggestion because of the deleted lines.)